### PR TITLE
Fix gaslift 1

### PR DESCRIPTION
**Pull request by totto82**
This PR is a first of several that aims to fix the gaslift code for cases with complex group controls with liquid / water constrains.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by svenn-t** [comment]
I have run the gaslift test cases in [opm-tests](https://github.com/OPM/opm-tests/tree/master/gaslift) with changes in this PR and seen improved matches in the comparison. With future changes to the code, based on this PR, many test cases will match (almost) exactly.

File: opm/simulators/wells/GasLiftSingleWellGeneric.cpp [hunk 1]
```diff
@@ -666,6 +666,13 @@ getRateWithLimit_(Rate rate_type, const BasicRates& rates) const
     //   for why the rate was limited.
     std::optional<Rate> target_type;
 
+    Scalar rate2;
+    if (rate_type == Rate::oil) {
+        rate2 = getRate_(Rate::water, rates);
+    } else {
+        rate2 = getRate_(Rate::oil, rates);
+    }
+
```
Comment (svenn-t) [inline opm/simulators/wells/GasLiftSingleWellGeneric.cpp:675]: `rate2` is only really used with `Rate::oil` and `Rate::water` so perhaps a comment here to indicate that for future reference? Could also initialize `rate2 = 0.0` and change the `else` to `else if (rate_type == Rate::water)` to indicate this further, but I leave that up to you.

**Review by svenn-t — COMMENTED** [review]
I have looked at the parts I have not contributed myself, and it looks good. Only one comment that you can take or leave.

**Comment by totto82** [comment]
jenkins build this please

File: opm/simulators/wells/GasLiftSingleWellGeneric.cpp (hunk 1 above)
Comment (totto82) [reply opm/simulators/wells/GasLiftSingleWellGeneric.cpp]: I have updated accordingly.

**Comment by totto82** [comment]
jenkins build this please

**Pull request merged**